From bed7636621589c139ed8d83842df3ce438b8493f Mon Sep 17 00:00:00 2001
From: Chirag Shah <chirag@nvidia.com>
Date: Mon, 17 Jun 2024 13:58:03 -0700
Subject: [PATCH] bgpd: backpressure - fix evpn route sync to zebra

In scaled EVPN + ipv4/ipv6 uni route sync to zebra,
some of the ipv4/ipv6 routes skipped reinstallation
due to incorrect local variable's stale value.

Once the local variable value reset in each loop
iteration all skipped routes synced to zebra properly.

Ticket: #3948828

Signed-off-by: Rajasekar Raja <rajasekarr@nvidia.com>
Signed-off-by: Chirag Shah <chirag@nvidia.com>

diff --git a/bgpd/bgp_zebra.c b/bgpd/bgp_zebra.c
index 5d5525156b..278e228d66 100644
--- a/bgpd/bgp_zebra.c
+++ b/bgpd/bgp_zebra.c
@@ -1801,6 +1801,8 @@ static void bgp_handle_route_announcements_to_zebra(struct thread *e)
 	bool install;
 
 	while (count < ZEBRA_ANNOUNCEMENTS_LIMIT) {
+		is_evpn = false;
+
 		dest = zebra_announce_pop(&bm->zebra_announce_head);
 
 		if (!dest)
-- 
2.43.2

